New version(2.7.0) compile time issue in android | Keeps app crashing

React native version = 0.70.14

compileOptions {
        sourceCompatibility JavaVersion.VERSION_18
        targetCompatibility JavaVersion.VERSION_18
    }



Stack trace

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-picker_picker:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':react-native-picker_picker:compileDebugJavaWithJavac'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:142)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:140)




ERROR  Invariant Violation: requireNativeComponent: "RNCAndroidDialogPicker" was not found in the UIManager.

This error is located at:
    in RNCAndroidDialogPicker
    in PickerAndroid (created by Picker)

--- a/android/src/main/java/com/reactnativecommunity/picker/ReactPickerLocalData.java
+++ b/android/src/main/java/com/reactnativecommunity/picker/ReactPickerLocalData.java
@@ -1,6 +1,15 @@
 package com.reactnativecommunity.picker;
 
-public record ReactPickerLocalData(int height) {
+public class ReactPickerLocalData {
+    private final int height;
+
+    public ReactPickerLocalData(int height) {
+        this.height = height;
+    }
+
+    public int getHeight() {
+        return height;
+    }
 
     @Override
     public boolean equals(Object o) {
@@ -8,6 +17,11 @@ public record ReactPickerLocalData(int height) {
         if (o == null || getClass() != o.getClass()) return false;
         ReactPickerLocalData that = (ReactPickerLocalData) o;
         return height == that.height;
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 + height;
     }
 
     @Override

--- a/android/src/main/java/com/reactnativecommunity/picker/ReactPickerShadowNode.java
+++ b/android/src/main/java/com/reactnativecommunity/picker/ReactPickerShadowNode.java
@@ -7,6 +7,6 @@ public class ReactPickerShadowNode extends LayoutShadowNode {
     @Override
     public void setLocalData(Object data) {
         Assertions.assertCondition(data instanceof ReactPickerLocalData);
-        setStyleMinHeight(((ReactPickerLocalData) data).height());
+        setStyleMinHeight(((ReactPickerLocalData) data).getHeight());
     }
 }

